b_cfn_elasticsearch_cloner/cloner_source/index.py

```diff
@@ -51,52 +55,85 @@ def handler(event: Dict[Any, Any], context: Any) -> None:
     :return: No return.
     """
 
-    logger.info("Starting processing DynamoDB events.")
+    logger.info('Starting processing DynamoDB events.')
+
+    use_embeddings = bool(SAGEMAKER_EMBEDDINGS_KEY and SAGEMAKER_ENDPOINT_NAME)
+
+    if not use_embeddings and any([SAGEMAKER_EMBEDDINGS_KEY, SAGEMAKER_ENDPOINT_NAME]):
+        raise OSError(
+            f'In order to use sentence embedding, all of the following enviroment variables are required: '
+            f'SAGEMAKER_ENDPOINT_NAME, SAGEMAKER_EMBEDDINGS_KEY. '
+            f'Else provide none of above.'
+        )
 
     # Send data to elasticsearch using bulk API.
     succeeded, failed = bulk(
         es,
-        dynamodb_to_es_generator(event),
+        dynamodb_to_es_generator(event, use_embeddings),
         stats_only=True,
         raise_on_error=False,
         raise_on_exception=False,
     )
 
-    logger.info(f"Finished processing DynamoDB events. Succeeded: {succeeded}, failed: {failed}")
+    logger.info(f'Finished processing DynamoDB events. Succeeded: {succeeded}, failed: {failed}')
+
+
+def get_sm_features(source: str, sagemaker_client: Any, endpoint_name: str) -> Dict[str, Any]:
+    sagemaker_response = sagemaker_client.invoke_endpoint(
+        EndpointName=endpoint_name,
+        Body=source.encode(encoding='utf-8'),
+        ContentType='text/plain'
+    )
+    source_features = json.loads(sagemaker_response['Body'].read().decode())
+    return source_features
+
+
+def get_embeddings(source: str) -> List[float]:
+    logger.info(f'Performing embeddings.')
+    source_features = get_sm_features(
+        source, sagemaker_client=sagemaker_client, endpoint_name=SAGEMAKER_ENDPOINT_NAME)
+    embeddings = source_features[SAGEMAKER_EMBEDDINGS_KEY]
+    return embeddings
 
 
-def dynamodb_to_es_generator(event: Dict[Any, Any]) -> Dict[Any, Any]:
+def dynamodb_to_es_generator(
+        event: Dict[Any, Any], use_embeddings: bool = False) -> Generator[Dict[str, Any], None, None]:
```

Comment:
list def function arguments vertically if there is a need to write in the next line

```python
def dynamodb_to_es_generator(
        event: Dict[Any, Any], 
        use_embeddings: bool = False
) -> Somethig:
```